src/prefect/engine/task_runner.py

```diff
@@ -167,9 +167,9 @@ def initialize_run(  # type: ignore
             task_run_count=run_count,
             task_name=self.task.name,
             task_tags=self.task.tags,
-            task_slug=self.task.slug,
         )
         context.setdefault("checkpointing", config.flows.checkpointing)
+        context.setdefault("task_slug", self.task.slug)
```

Comment:
Under what circumstances would `task_slug` not be set above in the flow runner? Alternatively, why should `task_slug` be set in the flow runner instead of always in the task runner?

Reply:
Your thoroughness is impressive.  With the new slugs mechanism, slugs are _only_ reliable if the Flow provides them so I actually should remove this and adjust the tests accordingly.